httpserver/service_node.h

```diff
@@ -112,13 +114,15 @@ class ServiceNode {
     /// (called when our old node got dissolved)
     void salvage_data() const;
 
-    void
-    attach_signature(const std::vector<std::string>& data,
-                     std::vector<std::shared_ptr<request_t>>& batches) const;
+    void attach_signature(std::shared_ptr<request_t>& request,
+                          const signature& sig) const;
 
     /// used on push and on swarm bootstrapping
     void relay_data(const std::shared_ptr<request_t>& req,
                     const sn_record_t& address) const;
+    void
+    relay_messages(const std::vector<service_node::storage::Item>& messages,
```

Comment:
I find the new API a bit confusing with `relay_data` and `relay_messages`. The former only relays, while the latter serialises, converts to requests and relays. How about we put signing code inside `to_requests`?

Reply:
Renamed as per offline discussion